### PR TITLE
DEV-3952 joker-num

### PR DESCRIPTION
**Pull request by mikysett**
New exercise for scripting piscine, devops branch

### CODE REVIEW

File: subjects/devops/joker-num/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+- `read`: this command can be used to read input from the command line.
+
+For example:
+
+```console
+$ read guess
+```
+
+After running that line, you would type your password, press `Enter`, and it would be stored in the `$password` variable for later use.
```
Comment (nprimo) [inline subjects/devops/joker-num/README.md:77]: after running `prettier` I got this diff:

```
diff --git a/subjects/devops/joker-num/README.md b/subjects/devops/joker-num/README.md
index d2053088..0cfc2192 100644
--- a/subjects/devops/joker-num/README.md
+++ b/subjects/devops/joker-num/README.md
@@ -7,6 +7,7 @@ The script will receive the number to guess as argument. The number should be be
 To achieve this you will have to use a `for` loop.

 When guessing a valid number the output will be:
+
 - Number bigger than the secret one: `Go down`.
 - Number smaller than the secret one: `Go up`.
 - Number equal than the secret one: `Congratulations, you found the number!`.
@@ -38,6 +39,7 @@ $
 - The player writing something that is not a valid number as an answer: The program will ask again to enter a guess and the number of tries won't be decremented.

 Here an example for the last scenario:
+
 ```console
 $ ./joker-num.sh 100
 Enter your guess (5 tries left):

```

File: sh/tests/solutions/joker-num.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+if [[ $# != 1
+    || -z "$1"
```
Comment (nprimo) [inline sh/tests/solutions/joker-num.sh:4]: Nice touch! Even if it's not needed to pass this exercise, could be nice to add this to the `hints` part of the subject

File: sh/tests/joker-num_test.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
```
Comment (nprimo) [inline sh/tests/joker-num_test.sh:2]: To make sure the tester exit at the first failed test, add the following:
```bash
set -e 
```

the `-o pipefail` should not add any value in this case as the `-u` for undefined variables

**Review by nprimo — CHANGES_REQUESTED** [review]
- Check `prettier` in the subject 
- One small change for the tests 
- A suggestion for a potential additional `hint` to add in the subject

File: subjects/devops/joker-num/README.md (hunk 1 above)
Comment (mikysett) [reply subjects/devops/joker-num/README.md:77]: done

File: sh/tests/solutions/joker-num.sh (hunk 2 above)
Comment (mikysett) [reply sh/tests/solutions/joker-num.sh:4]: done

File: sh/tests/joker-num_test.sh (hunk 3 above)
Comment (mikysett) [reply sh/tests/joker-num_test.sh:2]: I've added the full unofficial strict mode syntax

File: sh/tests/joker-num_test.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    expected=$(
+        ./solutions/joker-num.sh $args <<EOF
+$input
+EOF
+    )
+    diff <(echo "$submitted") <(echo "$expected")
+}
+
+# Good input, win
+input="1
```
Comment (nprimo) [inline sh/tests/joker-num_test.sh:26]: this variable is not used

Comment (mikysett) [reply sh/tests/joker-num_test.sh]: removed

**Pull request merged**